projects/cdk/components/dialog-host/dialog-host.component.ts

```diff
@@ -13,10 +14,12 @@ import {merge, Observable} from 'rxjs';
     animations: [TUI_PARENT_ANIMATION],
 })
 export class TuiDialogHostComponent<T extends TuiAriaDialogContext> {
-    readonly dialogs$ = merge(...this.dialogs);
+    readonly dialogs$ = combineLatest(this.allTypesDialogs).pipe(
+        map(allTypesDialogs => new Array<T>().concat(...allTypesDialogs)),
```

Comment:
i've understood right now that it is bad solution: it doesn't take order of dialog opening into consideration =/

I move PR to draft and think more

Reply:
fixed